win32ss/user/ntuser/defwnd.c

```diff
@@ -792,7 +792,7 @@ IntDefWindowProc(
                co_IntSendMessage(UserHMGetHandle(Wnd), WM_CONTEXTMENU, (WPARAM)UserHMGetHandle(Wnd), MAKELPARAM(-1, -1));
             }
          }
-         if (IS_KEY_DOWN(gafAsyncKeyState, VK_LWIN) || IS_KEY_DOWN(gafAsyncKeyState, VK_RWIN))
+         if (g_bWindowSnapEnabled && (IS_KEY_DOWN(gafAsyncKeyState, VK_LWIN) || IS_KEY_DOWN(gafAsyncKeyState, VK_RWIN)))
          {
             HWND hwndTop = UserGetForegroundWindow();
```

Comment:
I can confirm code equivalency.
You can make below code a newly-defined function.

Reply:
Yes, I could do that and I just experimented with it. But I do like the current version more, especially because the func is only and will only ever be called once, and the current version makes the differences in the key-handling more visible.
I do think just using a **block** in the current function is superior and allows more flexibility at a more flat callstack.

But an aspect of your suggestion is very good, and I do think that part is what motivated your suggestion:
I can move the declaration of the 3 Window-SnapOnly-variables "IstTaskBar, Style, Exstyle" into the Window-Snap-block.
It limits their scope and it keeps all parts of the WindowSnapping in this func within the Window-Snap-block!
It builds also with VC2010, I double-checked that. **Thanks for the input!** I will push that with commit3.

Reply:
Suggestion Resolved?

Reply:
Ok.